### PR TITLE
Generics Cleanup

### PR DESCRIPTION
**Pull request by Elscrux**
Attempt to fix most of the issues with raw generics.
`SubRoutinePool` still has issues with casting to the right `ProblemControllerProvider`. If you have an idea how to fix that, let me know @schweikart.

### CODE REVIEW

**Review by schweikart — CHANGES_REQUESTED** [review]

File: src/main/java/edu/kit/provideq/toolbox/featuremodel/anomaly/MetaSolverFeatureModelAnomaly.java [hunk 1]
```diff
@@ -14,16 +14,20 @@
  * Simple {@link MetaSolver} for FeatureModel problems.
  */
 @Component
-public class MetaSolverFeatureModelAnomaly extends MetaSolver<FeatureModelAnomalySolver> {
+public class MetaSolverFeatureModelAnomaly extends MetaSolver<
+        edu.kit.provideq.toolbox.featuremodel.anomaly.FeatureModelAnomalyProblem,
```
[the remark this answers is not in the document]
Comment (schweikart) [reply src/main/java/edu/kit/provideq/toolbox/featuremodel/anomaly/MetaSolverFeatureModelAnomaly.java]: same problem in line 29

**Comment by schweikart** [comment]
Can you rebase onto `develop`?

File: src/main/java/edu/kit/provideq/toolbox/featuremodel/anomaly/MetaSolverFeatureModelAnomaly.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Elscrux) [reply src/main/java/edu/kit/provideq/toolbox/featuremodel/anomaly/MetaSolverFeatureModelAnomaly.java]: Yeah, sorry I'm blind

**Comment by Elscrux** [comment]
@schweikart I rebased the branch

**Review by schweikart — APPROVED** [review]

**Pull request merged**